x-pack/plugins/monitoring/public/alerts/lib/get_alert_panels_by_category.tsx

```diff
@@ -10,73 +10,89 @@ import { EuiText, EuiToolTip } from '@elastic/eui';
 import { AlertPanel } from '../panel';
 import { ALERT_PANEL_MENU } from '../../../common/constants';
 import { getDateFromNow, getCalendar } from '../../../common/formatting';
-import { IAlertsContext } from '../context';
-import { AlertState, CommonAlertStatus } from '../../../common/types/alerts';
+import {
+  AlertState,
+  CommonAlert,
+  CommonAlertState,
+  CommonAlertStatus,
+} from '../../../common/types/alerts';
 import { PanelItem } from '../types';
 import { sortByNewestAlert } from './sort_by_newest_alert';
 import { Legacy } from '../../legacy_shims';
 
+interface MenuAlert {
+  alert: CommonAlert;
+  alertName: string;
+  states: CommonAlertState[];
+}
+interface MenuItem {
+  alertCount: number;
+  label: string;
+  alerts: MenuAlert[];
+}
 export function getAlertPanelsByCategory(
   panelTitle: string,
   inSetupMode: boolean,
   alerts: CommonAlertStatus[],
-  alertsContext: IAlertsContext,
   stateFilter: (state: AlertState) => boolean
 ) {
-  const menu = [];
-  for (const category of ALERT_PANEL_MENU) {
-    let categoryFiringAlertCount = 0;
-    if (inSetupMode) {
-      const alertsInCategory = [];
-      for (const categoryAlert of category.alerts) {
-        if (
-          Boolean(alerts.find(({ rawAlert }) => rawAlert.alertTypeId === categoryAlert.alertName))
-        ) {
-          alertsInCategory.push(categoryAlert);
-        }
-      }
-      if (alertsInCategory.length > 0) {
-        menu.push({
-          ...category,
-          alerts: alertsInCategory.map(({ alertName }) => {
-            const alertStatus = alertsContext.allAlerts[alertName];
```

Comment:
I don't see the point in using the alertsContext which has a list of the OOTB alerts to get the instance there since we fetch for the same data at an interval and it exists in the component state.